codegen/java-gen/src/main/scala/kalix/codegen/java/ViewServiceSourceGenerator.scala

```diff
@@ -229,12 +293,18 @@ object ViewServiceSourceGenerator {
       packageName,
       c"""$managedComment
          |
-         |public abstract class ${view.abstractViewName} extends $View<${view.state.messageType}> {
+         |public abstract class ${view.abstractViewName} extends $View {
          |  $emptyState
          |  $handlers
          |}
          |""",
       Nil)
   }
 
+  private def emptyStateMethodName(updateType: ProtoMessageType): String = {
+    val name = updateType.name
+    val suffix = if (name.endsWith("State")) "" else "State"
+    s"empty$name$suffix"
+  }
```

Comment:
Ah, not sure if I was looking at older commits or just missed it, this is good 👍